search-service/src/main/kotlin/com/egm/stellio/search/service/AttributeInstanceService.kt

```diff
@@ -37,16 +39,27 @@ class AttributeInstanceService(
     fun addAttributeInstances(
         temporalEntityAttributeUuid: UUID,
         attributeKey: String,
-        attributeValues: Map<String, List<Any>>
+        attributeValues: Map<String, List<Any>>,
+        parsedPayload: Map<String, Any>
     ): Mono<Int> {
         val attributeValue = getPropertyValueFromMap(attributeValues, NGSILD_PROPERTY_VALUE)
             ?: throw BadRequestDataException("Value cannot be null")
 
+        val instanceId = AttributeInstance.generateRandomInstanceId()
         val attributeInstance = AttributeInstance(
             temporalEntityAttribute = temporalEntityAttributeUuid,
+            instanceId = instanceId,
             observedAt = getPropertyValueFromMapAsDateTime(attributeValues, EGM_OBSERVED_BY)!!,
             value = valueToStringOrNull(attributeValue),
-            measuredValue = valueToDoubleOrNull(attributeValue)
+            measuredValue = valueToDoubleOrNull(attributeValue),
+            payload = Json.of(
+                extractAttributeInstanceFromParsedPayload(
+                    parsedPayload,
+                    attributeKey,
+                    null,
+                    instanceId
+                )
+            )
```

Comment:
not totally convinced by this... wondering how it could be improved / better abstracted?

Reply:
It could be done in the handler level, for each attribute we extract it's payload (only the default instance for now since multi instance is not supported yet) and we pass it to `addAttributeInstances`